articles/quickstart/native/android/04-user-profile.md

```diff
@@ -37,67 +37,57 @@ WebAuthProvider.init(auth0)
+The profile obtained this way is OIDC-conformant. Depending on the [scopes](/scopes/current) you requested when logging in, the profile contains different information. The result will never contain fields outside the OIDC specification.
+2. Get the user's full profile using the [Management API](/api/management/v2#!/Users). This step is explained next:
+
+
+Create an instance of the Users API client using the id token that you saved in the log in step. In the snippet bellow we obtain it from the extras that the LoginActivity class has passed when starting this activity. The API client is used to request the users' profile data.
 
 ```java
 // app/src/main/java/com/auth0/samples/activities/MainActivity.java
 
 Auth0 auth0 = new Auth0(this);
 auth0.setOIDCConformant(true);
 
-String idToken = CredentialsManager.getCredentials(this).getIdToken();
+String idToken = getIntent().getStringExtra(LoginActivity.KEY_ID_TOKEN);
 UsersAPIClient usersClient = new UsersAPIClient(auth0, idToken);
-AuthenticationAPIClient authClient = new AuthenticationAPIClient(auth0);
 ```
 
 ::: note
-Do not hardcode the Auth0 `domain` and `clientId` values. We recommend you add them to the `strings.xml` file.
+Do not hardcode the Auth0 `domain` and `clientId` values when creating the Auth0 instance. We recommend you add them to the `strings.xml` file.
 :::
 
-To get the user's information:
-1. Use the user's access token to call the `userInfo` method in the `AuthenticationAPIClient` client instance.
-You get an instance of the `UserProfile` profile. The profile is OIDC-conformant. Depending on the on the [scopes](/scopes/current) you requested, the profile contains different information. 
-2. To get the user's full profile, use the [Management API](/api/management/v2#!/Users).
+
+Decode the id token using a [JWT library](https://github.com/auth0/JWTDecode.Android) and obtain the Subject value, which represents the id of the user to whom the token was issued.
 
 ```java
 // app/src/main/java/com/auth0/samples/activities/MainActivity.java
 
-String accessToken = CredentialsManager.getCredentials(this).getAccessToken();
-authenticationClient.userInfo(accessToken)
-    .start(new BaseCallback<UserProfile, AuthenticationException>() {
-
-        @Override
-        public void onSuccess(final UserProfile userInfo) {
-            String userId = userInfo.getId();
-            // fetch the full user profile
-        }
-
-        @Override
-        public void onFailure(AuthenticationException error) {
-            //show error
-        }
-    });
+private String getUserId(String idToken) {
+    return new JWT(idToken).getSubject();
+}
 ```
 
-When you get the `sub` value, call the [Management API](https://auth0.com/docs/api/management/v2#!/Users).
-
-Use the `UsersAPIClient` client and the user's ID to get the full user profile.
+Use the user id to call `getProfile` on the Users API client and obtain the full user profile. Use the received data to update the layout.
```

Comment:
~Use the user id~
Use the `userId`

Reply:
I rather not use that as the claim is called "sub" and we referred to it in the above paragraph as "obtaining the ... user id"